File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/model/roomResult/EndResultModel.java
package com.syahputrareno975.cardbattlemodule.model.roomResult;

import com.syahputrareno975.cardbattlemodule.model.player.PlayerModel;
import com.syahputrareno975.cardbattlemodule.model.playerBattleResult.PlayerBattleResultModel;
import com.syahputrareno975.cardbattlemodule.model.roomReward.RoomRewardModel;
import java.io.Serializable;
import java.util.ArrayList;

public class EndResultModel  implements Serializable {
    public PlayerModel Winner = new PlayerModel();
    public ArrayList<PlayerBattleResultModel> allBattleResult  = new ArrayList<PlayerBattleResultModel>();
    public int FlagResult = 0;
    public RoomRewardModel Reward = new RoomRewardModel();

    public EndResultModel() {
    }

    public EndResultModel(PlayerModel winner, ArrayList<PlayerBattleResultModel> allBattleResult, int flagResult, RoomRewardModel reward) {
        Winner = winner;
        this.allBattleResult = allBattleResult;
        FlagResult = flagResult;
        Reward = reward;
    }
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/model/roomResult/EndResultModel.kt
package com.syahputrareno975.simplecardbattle.model.roomResult

import com.syahputrareno975.simplecardbattle.model.player.PlayerModel
import com.syahputrareno975.simplecardbattle.model.playerBattleResult.PlayerBattleResultModel
import com.syahputrareno975.simplecardbattle.model.roomReward.RoomRewardModel
import java.io.Serializable

class EndResultModel  : Serializable {
    var Winner : PlayerModel = PlayerModel()
    var allBattleResult : ArrayList<PlayerBattleResultModel> = ArrayList<PlayerBattleResultModel>()
    var FlagResult = 0
    var Reward : RoomRewardModel = RoomRewardModel()

    constructor()

    constructor(Winner: PlayerModel, allBattleResult: ArrayList<PlayerBattleResultModel>, FlagResult: Int, Reward: RoomRewardModel) {
        this.Winner = Winner
        this.allBattleResult = allBattleResult
        this.FlagResult = FlagResult
        this.Reward = Reward
    }
}